jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/ProcessServiceImpl.java

```diff
@@ -131,6 +133,39 @@ public Long startProcessFromNodeIds(String deploymentId, String processId, Corre
     }
 
 
+    @Override
+    public Map<String, Object> startSynchronousProcess(String deploymentId, String processId, Map<String, Object> params) {
+        DeployedUnit deployedUnit = deploymentService.getDeployedUnit(deploymentId);
+        if (deployedUnit == null) {
+            throw new DeploymentNotFoundException("No deployments available for " + deploymentId);
+        }
+        if (!deployedUnit.isActive()) {
+            throw new DeploymentNotActiveException("Deployment " + deploymentId + " is not active");
+        }
+
+        RuntimeManager manager = deployedUnit.getRuntimeManager();
+
+        params = process(params, ((InternalRuntimeManager) manager).getEnvironment().getClassLoader());
+        RuntimeEngine engine = manager.getRuntimeEngine(getContext(params));
+        KieSession ksession = engine.getKieSession();
+
+        Map<String, Object> vars = new HashMap<>(params); // copy variables
```

Comment:
Check that params is not null to avoid NPE:
```suggestion
        Map<String, Object> vars = params!=null ? new HashMap<>(params) : new HashMap<>();
```

Reply:
done